### DISCUSSION TITLE
Track wilderness loot key values and deaths on a customizable overlay or output to a text file?

### DISCUSSION THREAD

**Discussion by BreeNET** [runelite/runelite#17703] [Ideas]
Im willing to pay commission for this plugin as it would be pretty niche and not a lot of people would use it. So I imagine if this was picked up by anyone, it wouldn't be prioritized. Or maybe this already exists and I haven't found it yet. Let me know if commission isn't allowed to be advertised and I'll remove this part of my post.  

I'm a pking streamer. I have an overlay on my stream that shows gained and lost GP from pking for that day. 

Right now it's just a chat command. If I get a loot key from pking that's worth 4,264,881, a mod will type in +4.2m into my chat which then reflects on my overlay. If I die and the value of my death is 16,991,012 then mods will type -16.9m and it reflects on my overlay. 

I'm looking to simplify this with a plugin. Can it be done automatically via a plugin? When I die, take the value of my death that pops up in chat and round it to the nearest hundredth. 

When I open a loot key and it shows the value along with all the items in the loot key, take that value and round it to the nearest hundredth. 

Right now my overlay reads from 2 text files on my desktop. Kills.txt and deaths.txt. I'm assuming that could just be made into an overlay on runelite. Either would be fine. Write it to an overlay or write it to the 2 text files. I like the text file because the font and color and size can be customized so if the plugins overlay could allow customization that'd be cool too.

Now that I think about it,  I might actually prefer outputting the value to a text file because I'm not sure how large plugin overlay font can be. I have my current overlay set to size 32 font and it just rests on a single line right above my chat box. It doesn't get in the way of my own gameplay since it's just an overlay on my stream, not my actual runelite client.

`Gained: 32.7m       Lost: 89.04m`

Gained is green, lost is red. 

Another nice thing is that when I end my stream, my overlay automatically resets to 0. I don't need to manage it every single day or go in and clear anything or reset anything. Bonus points if the plugin can have a manual reset button just in case, but also automatically reset itself to 0 somehow when I close my client or when I'm logged out for more than 60 minutes but didn't close runelite.

**Comment by YvesW** [comment]
The Dink plugin should be able to send you that data. You can then probably have some script extract the values from the JSON and add that to the value in your file.

**Reply by BreeNET** [reply]
This is perfect. Ty

**Comment by BreeNET** [comment]
> The Dink plugin should be able to send you that data. You can then probably have some script extract the values from the JSON and add that to the value in your file.
